codemod/output/PutPart.ts

```diff
@@ -0,0 +1,338 @@
+import { SignedS3AWSRequest } from "./SignedS3AWSRequest";
+import { Global } from "./Global";
+import {
+  PARTS_MONITOR_INTERVAL_MS,
+  COMPLETE,
+  ABORTED,
+  PAUSED,
+  CANCELED,
+  EVAPORATING,
+  ERROR,
+  PAUSING
+} from "./Constants";
+
+//http://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadUploadPart.html
+class PutPart extends SignedS3AWSRequest {
+	public part: any;
+	public partNumber: any;
+	public start: any;
+	public end: any;
+	public stalledInterval: any;
+	public size: any;
+	public result: any;
+  static size: number;
+
+  constructor(fileUpload, part) {
+    super(fileUpload);
+
+    this.part = part;
+    this.partNumber = part.partNumber;
+    this.start = (this.partNumber - 1) * fileUpload.con.partSize;
+    this.end = Math.min(
+      this.partNumber * fileUpload.con.partSize,
+      fileUpload.sizeBytes
+    );
+
+    const request = {
+      method: "PUT",
+      path: `?partNumber=${this.partNumber}&uploadId=${fileUpload.uploadId}`,
+      step: `upload #${this.partNumber}`,
+      x_amz_headers:
+        fileUpload.xAmzHeadersCommon || fileUpload.xAmzHeadersAtUpload,
+      contentSha256: "UNSIGNED-PAYLOAD",
+      onProgress: this.onProgress.bind(this)
+    };
+
+    this.updateRequest(request);
+  }
+
+  getPartMd5Digest() {
```

Comment:
It might be worth migrating to the async await syntax, it's valid in other files.

Reply:
The same answer as below, this isn't within the scope of the PR, but I agree that would be much better.